src/components/magnet/magnet-canvas.tsx

```diff
@@ -98,19 +114,22 @@ export class MagnetCanvas extends BaseComponent<IProps, IState> {
       height
     };
 
+    const magnets = [magnet1, magnet2];
+    const magnetModels = simulation.magnets;
+
     return (
       <Stage options={options} style={{width, height}}>
         {
           simulation.showCloud &&
-          <GradientField magnets={[magnet1, magnet2]} width={width} height={height} cellSize={20} />
+          <GradientField magnets={magnets} magnetModels={magnetModels} width={width} height={height} cellSize={20} />
```

Comment:
I know we discussed just passing the models here and dropping the React state entirely, but I haven't had a chance to do that yet